### PR TITLE
Add systemd example

### PR DESCRIPTION
**Pull request by kfox1111**

### CODE REVIEW

File: examples/systemd/conf/agent/main.conf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    log_level = "DEBUG"
+    trust_domain = "example.org"
+    server_address = "localhost"
+    server_port = 8081
+
+    # Insecure bootstrap is NOT appropriate for production use but is ok for
+    # simple testing/evaluation purposes.
+    insecure_bootstrap = true
+
+    join_token = "cdf1885a-1db8-4a83-aa16-ad8c84761fa8"
```
Comment (sorindumitru) [inline examples/systemd/conf/agent/main.conf:11]: This is not the best, but I imagine you want to wait for something like that dns/http pop attestor being available, right?

File: examples/systemd/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+Edit /etc/spire/server/main.conf and update with settings as needed.
+
+Edit /etc/spire/agent/main.conf and update with settings as needed.
+
+Enable the main server:
+
+```
+systemctl enable spire-server@main
```
[the remark this answers is not in the document]
Comment (kfox1111) [reply examples/systemd/README.md:15]: Yeah. It lets you have multiple servers/agents. Either for nesting or redundancy. I do plan on using both.

File: examples/systemd/conf/agent/main.conf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        }
+    }
+
+    NodeAttestor "join_token" {
+        plugin_data {}
+    }
+
+    WorkloadAttestor "systemd" {
+        plugin_data {}
+    }
```
[the remark this answers is not in the document]
Comment (kfox1111) [reply examples/systemd/conf/agent/main.conf:27]: Yes, but I think systemd is a better plugin then unix as a default suggestion. Many different services run as root (ssh, apache, kubelet, etc). But each would probably want its own identity. The systemd attestor can tell them apart.

File: examples/systemd/conf/agent/main.conf (hunk 1 above)
Comment (kfox1111) [reply examples/systemd/conf/agent/main.conf:11]: yeah, that would be a better default once its available.

This is just an example config file really. Its expected an end user will need to update it to use it properly.

File: examples/systemd/system/spire-agent@.service [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+Before=remote-fs-pre.target spire-agent.target
+Wants=network-online.target local-fs.target time-sync.target remote-fs-pre.target spire-agent.target
+
+[Service]
+WorkingDirectory=/var/lib/spire/agent/%i
+StateDirectory=spire/agent/%i
+RuntimeDirectory=spire/agent/sockets/%i
+RuntimeDirectoryPreserve=true
+ConfigurationDirectory=spire/agent
+ExecReload=/bin/kill -HUP $MAINPID
```
[the remark this answers is not in the document]
Comment (kfox1111) [reply examples/systemd/system/spire-agent@.service]: Fixed.

File: examples/systemd/system/spire-server.target [hunk 5]
```diff
@@ -0,0 +1,8 @@
+[Unit]
+Description=SPIRE Server target allowing to start/stop all spire-server@.service instances at once
+PartOf=spire.target
+Before=ceph.target
```
[the remark this answers is not in the document]
Comment (kfox1111) [reply examples/systemd/system/spire-server.target]: Fixed

File: examples/systemd/README.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+```
+
+Start the main server:
+
+```
+systemctl enable spire-server@main
+```
+
+Enable the main agent:
```
[the remark this answers is not in the document]
Comment (kfox1111) [reply examples/systemd/README.md:30]: Updated.

File: examples/systemd/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
+To install, download the newest spire-server and spire-agent binaries from the SPIRE website and place in /bin
```
Comment (amartinezfayo) [inline examples/systemd/README.md:1]: I think that it would be nice to have a title for this document.

**Review by amartinezfayo — COMMENTED** [review]
Thank you @kfox1111 for this contribution, it looks great!
Can we update the README.md file in the root folder to include a link with this example along with a short description as we have for the other examples?

**Comment by kfox1111** [comment]
Is the test failure related?

**Comment by amartinezfayo** [comment]
>Is the test failure related?

No, the failure is in the Envoy example.

**Review by amartinezfayo — APPROVED** [review]

**Pull request merged**